test/NuGet.Core.Tests/NuGet.Protocol.Tests/RawSearchResourceTests.cs

```diff
@@ -38,8 +38,22 @@ public async Task RawSearchResource_SearchEncoding()
                 SupportedFrameworks = new string[] { ".NETFramework,Version=v4.5" }
             };
 
+            var skip = 0;
+            var take = 1;
             // Act
-            var packages = await searchResource.Search("azure b", searchFilter, 0, 1, NullLogger.Instance, CancellationToken.None);
+            var packages = await searchResource.Search(
+                    (_client, uri) => httpSource.ProcessHttpStreamAsync(
+                        new HttpSourceRequest(uri, Common.NullLogger.Instance),
+                        s => searchResource.ProcessHttpStreamTakeCountedItemAsync(s, take, CancellationToken.None),
+                        NullLogger.Instance,
+                        CancellationToken.None),
```

Comment:
This test already has a `TestHttpSource` abstraction, passed into the constructor of `RawSearchResourceV3`. I don't understand why this lamba function is needed. It seems to make the API more difficult to understand.

Reply:
Ok. Removed it from here.